packages/api/src/external/aws/address.ts

```diff
@@ -1,39 +1,82 @@
-import { Address } from "../../domain/medical/address";
 import {
-  getCoordinatesFromLocation,
+  CoordinatesAndRelevance,
+  getCoordinatesAndRelevanceFromLocation,
+  getLocationResultPayload,
   makeLocationClient,
 } from "@metriport/core/external/aws/location";
-import { Config } from "../../shared/config";
 import * as AWS from "aws-sdk";
-import { Coordinates } from "@metriport/core/external/aws/location";
+import { Address } from "../../domain/medical/address";
+import { Config } from "../../shared/config";
+
+export type AddressAndRelevance = {
+  address: Address;
+  relevance: number;
+};
 
 const indexName = Config.getPlaceIndexName();
 const placeIndexRegion = Config.getPlaceIndexRegion();
 const client = makeLocationClient(placeIndexRegion);
 
+export function buildAddressText(address: Address): string {
+  return `${address.addressLine1}, ${address.city}, ${address.state} ${address.zip}`;
+}
+
 /**
- * Geocodes a list of addresses using Amazon Location Services.
- * @param addresses
- * @returns
+ * Adds coordinates to addresses that don't already have them.
+ *
+ * @param addressList a list of Address objects.
+ * @returns a list of updated Address objects and their relevance score from geocoding.
  */
-export async function geocodeAddresses(addresses: Address[]): Promise<Coordinates[]> {
-  const resultPromises = await Promise.allSettled(
-    addresses.map(async address => {
-      const addressText = `${address.addressLine1}, ${address.city}, ${address.state} ${address.zip}`;
-      const countryFilter = address.country ?? "USA";
-
-      const params: AWS.Location.Types.SearchPlaceIndexForTextRequest = {
-        Text: addressText,
-        MaxResults: 1,
-        Language: "en",
-        FilterCountries: [countryFilter],
-        IndexName: indexName,
-      };
-
-      const locationResponse = await client.searchPlaceIndexForText(params).promise();
-      return getCoordinatesFromLocation({ result: locationResponse });
+export async function addGeographicCoordinates(
+  addressList: Address[]
+): Promise<AddressAndRelevance[]> {
+  const geocodingUpdates = await Promise.allSettled(
+    addressList.map(async address => {
+      if (address.coordinates) {
+        return;
+      }
```

Comment:
This is being called by patient update as well, right? Then, shouldn't we also update the coordinates then?

Reply:
Unless I'm misunderstanding what you mean, we do that on patient update

Reply:
Oh you mean if they update the existing address, we should update the existing coordinates? 
When cx create or update patients, they are not expected to add coordinates (dash doesnt even support that), so all new or updated addresses will not have this field (this is tested thru patient edit on dashboard). 

However, it is possible that cx creates a patient and passes coordinates for the address - this is unlikely, but not impossible, tho we provide no documentation for what the fields are and how they could do that. Perhaps, I could add logic to drop cx-provided coordinates from addresses?

Reply:
No, I mean, on the patient update route we call a function that calls this:

https://github.com/metriport/metriport/pull/1323/files/6f9cb46c6da9d0faaadbe39b605defd120c00310#diff-8c5ee5756e85f5adfecd084e785eac27b1a243484d5a2be90386755eef9a2097R124

Reply:
Which I see now were [moved to the command](https://github.com/metriport/metriport/pull/1323/files#diff-ed9bffa17b44ed55fd85ecc24fcd1e9cab83eb5f2d1e6904cdc906f116d50095R30)

Reply:
Also:

> When cx create or update patients, they are not expected to add coordinates

If I understand correctly, this function adds coordinates to an address w/o coordinates, so why are you referring to cxs adding address w/ coordinates? Am I misunderstanding something?

Reply:
I just mean that this function adds coordinates to addresses that don't have them. But in theory, it's possible for CX to add a patient with coordinates already included in the address payload if they use the API. In which case, we won't be checking if those are accurate. I think we could implement a couple of approaches to avoid this problem, altho it's unlikely to happen. 

One approach would be to change the PatientCreateCmd schema to not allow the coordinates on the address.
Alternatively, we could just strip it off on patient create/update in case the CX adds it. In both scenarios, we will make sure that we add the coordinates ourselves based on the provided addresses